DynamoDS/Dynamo
# MAGN-8245 Upgrade Helix to .net 4.5

### Purpose

This PR does the following.

- Integrates a version of the Helix Toolkit component which is now the standard. Previously, we had used a custom build of the Helix Toolkit. See [MAGN-8245](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8245), for a list of our contributions to that project for this purpose.
- Fixes [MAGN-8245](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8245), by integrating the new version of the Helix Toolkit using .net 4.5, which supports software fallback (WARP) for rendering over Remote Desktop.
- Replaces the Helix Toolkit and SharpDX assembly references to the `\extern` folder with NuGet packages representing the same.

### Impact assessment

- [ ] Changes the way 3rd party nodes work (Add ikeough to the reviewer list)
- [ ] Includes migrations (Add ikeough to the reviewer list)
- [ ] Changes a public interface (Add lukechurch to the reviewer list)
- [ ] Affects which components of the system depends on which other components (Add pboyer to the reviewer list)
- [x] Introduces new components (Add lukechurch to the reviewer list)
- [ ] Has a performance impact (Add keyu to the reviewer list)
- [x] Changes the installer (Add sharadkjaiswal to the reviewer list)
  - The `d3d10warp.dll` and `D3DCompiler_44.dll` files must now be bundled with the installer.

#### References

- See [MAGN-8245](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8245) for a list of PRs 

#### Testing strategy

- Ensure 3D graphics are visible on Remote Desktop.
- Ensure 3D graphics are visible on machines which do not support DirectX11.
- Ensure 3D graphics are available on machines with "switchable" graphics set to the CPU graphics controller (I.e. Intel Integrated Graphics).

### Review staging

- [ ] Architectural Design Review
- [ ] UI Review

### Reviewers

@pboyer 

### FYIs

@jnealb @sharadkjaiswal
